lib/rspec/core/formatters/json_formatter.rb

```diff
@@ -6,17 +6,30 @@ module Core
     module Formatters
       # @private
       class JsonFormatter < BaseFormatter
-        Formatters.register self, :message, :dump_summary, :dump_profile, :stop, :close
+        Formatters.register self, :message, :dump_summary, :dump_profile, :stop,
+                            :close, :example_group_started, :example_group_finished
 
         attr_reader :output_hash
 
         def initialize(output)
           super
+          @start = Hash.new(0)
+          @execution_times = Hash.new(0)
           @output_hash = {
             :version => RSpec::Core::Version::STRING
           }
         end
 
+        def example_group_started(group)
+          key =  group.group.metadata[:location]
```

Comment:
* The use of `group` for the argument here confused me since it caused you to use `group.group`.  Then I remembered it's a notification object.  I think it would be easier to understand if the arg was called `notification` and you used `notification.group`.
* `:location` isn't a sufficiently unique key.  When shared example groups are involved you can have a location for which multiple groups are defined.  Or the same thing can happen if users define examples or groups in a loop.  `ExampleGroup#id` (new since 3.2) is guaranteed unique for every example group.  (That's why it exists!)  However, I think that you can probably use the group object itself as the key in the hash here -- no need to get string identifier for it.